lib/script/script.js

```diff
@@ -3112,6 +3134,11 @@ class Script {
       const raw = stack.pop();
       const redeem = Script.fromRaw(raw);
 
+      if (!(flags & Script.flags.VERIFY_SEGWIT_RECOVERY) !== 0
```

Comment:
you need to check for the flag to be set, the `!` is doing the opposite. Also the reason for some failing tests.

Reply:
You're the man , thanks getting rid of that symbol corrects all the failing tests for Segwit Recovery.